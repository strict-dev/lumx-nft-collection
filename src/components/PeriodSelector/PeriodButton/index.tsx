import { periods } from '..'

type PeriodButtonProps = {
  label: typeof periods[number]
  selected: boolean
  setSelected: (value: string) => void
}

const PeriodButton: React.FC<PeriodButtonProps> = ({
  label,
  selected,
  setSelected
}) => {
  return (
    <button
      className={`px-2 py-1 text-sm font-bold transition rounded-sm 
      ${selected && 'brand-gradient text-white'}
      text-neutral-600 
      hover:brightness-110`}
      onClick={() => setSelected(label)}
    >
      {label.toUpperCase()}
    </button>
  )
}

export default PeriodButton
