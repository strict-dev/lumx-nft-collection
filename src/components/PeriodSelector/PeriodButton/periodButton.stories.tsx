import { ComponentMeta, ComponentStory } from '@storybook/react'

import PeriodButton from '.'

export default {
  title: 'PeriodButton',
  component: PeriodButton
} as ComponentMeta<typeof PeriodButton>

const Template: ComponentStory<typeof PeriodButton> = (args) => (
  <PeriodButton {...args} />
)

export const Default = Template.bind({})

Default.args = {
  label: '1m',
  selected: true
}
