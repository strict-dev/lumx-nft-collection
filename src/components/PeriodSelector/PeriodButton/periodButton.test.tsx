import { render } from '@testing-library/react'

import PeriodButton from '.'

describe('<PeriodButton />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <PeriodButton
        label="1d"
        selected={true}
        setSelected={() => {
          '1'
        }}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
