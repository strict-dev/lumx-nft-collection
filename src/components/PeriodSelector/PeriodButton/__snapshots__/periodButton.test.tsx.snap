// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeriodButton /> should render the heading 1`] = `
<button
  class="px-2 py-1 text-sm font-bold transition rounded-sm 
      brand-gradient text-white
      text-neutral-600 
      hover:brightness-110"
>
  1D
</button>
`;
